pkgs/data/fonts/joypixels/default.nix

```diff
@@ -1,24 +1,95 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, acceptLicense ? false }:
+
+let inherit (stdenv.hostPlatform) system;
+
+    throwSystem = throw "Unsupported system: ${system}";
+
+    systemTag = {
+      x86_64-darwin = "nix-darwin";
+      x86_64-linux = "nix-os";
+    }.${system} or throwSystem;
+
+    capitalized = {
+      x86_64-darwin = systemTag;
+      x86_64-linux = "NixOS";
+    }.${system} or throwSystem;
+
+    fontFile = {
+      x86_64-darwin = "Apple%20Color%20Emoji.ttc";
+      x86_64-linux = "joypixels-android.ttf";
+    }.${system} or throwSystem;
+
+    name = {
+      x86_64-darwin = "joypixels-apple-color-emoji.ttc";
+      x86_64-linux = fontFile;
+    }.${system} or throwSystem;
+
+    ext = {
+      x86_64-darwin = "ttc";
+      x86_64-linux = "ttf";
+    }.${system} or throwSystem;
+
+    joypixels-free-license = {
+      spdxId = "LicenseRef-JoyPixels-Free-6.0";
+      fullName = "JoyPixels Free License Agreement 6.0";
+      url = "https://cdn.joypixels.com/distributions/${systemTag}/license/free-license.pdf";
+    };
+    joypixels-license-appendix = {
+      spdxId = "LicenseRef-JoyPixels-NixOS-Appendix";
+      fullName = "JoyPixels ${capitalized} License Appendix";
+      url = "https://cdn.joypixels.com/distributions/${systemTag}/appendix/joypixels-license-appendix.pdf";
+    };
+
+    sha256 = {
+      x86_64-darwin = "043980g0dlp8vd4qkbx6298fwz8ns0iwbxm0f8czd9s7n2xm4npq";
+      x86_64-linux = "1vxqsqs93g4jyp01r47lrpcm0fmib2n1vysx32ksmfxmprimb75s";
+    }.${system} or throwSystem;
+
+in
+
+assert !acceptLicense -> throw ''
```

Comment:
This is a bit abusive of an `assert`, since it's not really made for this. But I guess it works, and it's pretty simple, so why not

Reply:
I'm open to a better way to do this. All I want is a way to visually warn a user and instruct them on how to fix the issue.

Reply:
Could we at least move it to `src` attribute so that `joypixels.override { acceptLicense = true; }` works?

Reply:
I'm fine with this change. Could you explain how passing `joypixels.override { acceptLicense = true; }` differs from specifying it in the overlay like this `super.callPackage joypixels/default.nix { acceptLicense = true; };`.

I expected the overlay to work the same way an override would so I didn't stumble on this.

Reply:
`makeOverridable` checks the type of the package expression after passing the original arguments:

https://github.com/NixOS/nixpkgs/blob/11af562a96f22dde27d20dc23604bde43964dd2b/lib/customisation.nix#L67-L81

so if the assertion fails you are out of luck.

If you pass the `acceptLicense = true` to the original arguments, then it cannot complain.

Reply:
I *could* use some advice on formatting the nix expression. The error message should be formatted for 80 columns imo, but, as it is the indentation causes it to overflow when editing the expression which is a bit confusing.

Should I reformat overflowing urls using string concatenation?

Reply:
IMO, overflowing lines is still less pain than joining paths. Most editors can visually mark soft-wrapped lines just fine.